### ISSUE TITLE
Addresses as nodes outside the building outline are valid in Spain

### ISSUE THREAD

**Issue by javiersanp** [streetcomplete/StreetComplete#3162]
**Use case 1**

The addresses quest currently present all buildings without address as errors.  In Spain or Italy, the recommendation for address is to put them not in the area of the building or parcel but in nodes in the outline. Please see this [mall](https://www.openstreetmap.org/way/403764654) with many addresses in [node1](https://www.openstreetmap.org/node/7160299351), [node2](https://www.openstreetmap.org/node/7160299306) and [node3](https://www.openstreetmap.org/node/7160299307) in two different streets.

**Use case 2**

For detached buildings, the nodes could also be in the surrounding area instead of a building outline. See this [example](https://www.openstreetmap.org/way/395543628). Frequently presents in area amenities such as schools,  hospitals. 

Both cases results in numerous false errors. When users try to fix them, SC put it in the building, resulting in duplicated information.

**Proposed Solution**

The preferred solution for case 1 could be to detect buildings with addr:* keys in any of it chield nodes to exclude them as errors.

For the case 2. As I know it's difficult task to find the related addresses, you can consider to prevent the duplication. When the user fixs the address, check for nearby elements with the same data that are not amenities, shops or any other feature by themselves. Give a warning preventing to input a duplicate.

Exclude from this check features with keys like amenity, shop, man_made, landuse, historic and many others because it's fine to have a duplicated address.

The main candidates are nodes with only addr:*, entrance=yes or barrier=gate.

In any case, you can check also ways/relations with the building key and the same addr as this would be a duplicate.

Related issue #3056

**Comment by matkoniecz** [comment]
Is https://www.openstreetmap.org/way/395543628 really considered as a valid tagging? Is it documented somewhere at OSM Wiki or in other such place?

**Comment by javiersanp** [comment]
ad 1) No, I can't. I was speaking without currently test it in the app so it's fine if SC already handle this.

ad 2). Yes, it's documented [here](https://wiki.openstreetmap.org/wiki/Addresses). "Tags can be added to : ...     nodes that are parts of building polygons ( = entrance=*s) or on the access to a property (e.g. on barrier=gate or barrier=entrance), used in places where addresses are assigned to entrances and accesses rather than to buildings)". 

Although there is no consensus on how to handle buildings with multiple house numbers, the multiple nodes approach figures in the list of possibilities. It's the preferred solution in Spain where multiple house number plates could be present in a building/property (map what's on the ground). Note that multiple addresses could be in different streets so the option of separate them with comas in the addr:housenumber tag could not be applied.

It's a pity to loose all the housenumber quest if the only false errors are the detached buildings (address not in the building outline but in the property access). Maybe tagging could be improved? For example a relation to link the building with the address nodes.

**Comment by matkoniecz** [comment]
> Yes, it's documented here.

That is general info about all approaches that are used somewhere. Is it documented somewhere that such tagging is valid specifically in Spain?

https://github.com/streetcomplete/StreetComplete/blob/master/app/src/main/java/de/westnordost/streetcomplete/quests/address/AddHousenumber.kt#L28 - Italy is disabled already

**Comment by javiersanp** [comment]
Sorry but I can't fully understand your request. If its valid somewhere its valid also in Spain. Could you please give me refrences for such specific documentation in any place to take them as examples?

The topic of house numbers arise many times on the Talk-es mailing list particularly in the context of the Cadastre import. House numbers are been mapped alternatively in nodes or areas, but it's preferred the first over the second (see [here](https://wiki.openstreetmap.org/wiki/ES:Directrices_de_etiquetado_espa%C3%B1olas#N.C3.BAmeros_de_portal)).

I can't find documented the specific topic of multiple house numbers for an area, but the node solution is the only one I know for this. You can expect to find it applied in this [places](https://wiki.openstreetmap.org/wiki/Spanish_Cadastre/Buildings_Import/Results).

IMO locally disabling the quest is the easy way for technical reasons but we loose the valuable data that user could complete using the app. It would be better to extend the checking for nearby duplicates when the building is detached. A duplicated address is an error in any case.

**Comment by matkoniecz** [comment]
> If its valid somewhere its valid also in Spain.

Note "used in places where addresses are assigned to entrances and accesses rather than to buildings" qualifier.

> Could you please give me refrences for such specific documentation in any place to take them as examples?

https://wiki.openstreetmap.org/wiki/ES:Directrices_de_etiquetado_espa%C3%B1olas#N.C3.BAmeros_de_portal is exactly something that I was looking for - instructions to mappers that describe recommended tagging scheme that makes assumptions made by SC invalid

> It would be better to extend the checking for nearby duplicates when the building is detached.

But it is only possible to do this after user entered address. So after quest was asked and asnwered app would detect that it should not be asked in the first place and nothing would be tagged. Then it would be asked again, and again and again (even if it would be recorded locally that this is invalid target - other mappers would be asked).

In other words

>  Any answer the user can give must have an equivalent tagging (Quest should not reappear to other users when solved by one)

quest requirement would be impossible to satisfy. Or is there some tag to indicate "this building has address nodes placed outside it"? This would allow to at least theoretically rescue this quest.

**Comment by matkoniecz** [comment]
In summary:

StreetComplete works fine with address tagged on area with buildings inside 

StreetComplete works fine with address tagged on nodes inside building or on outline of a building.

Address quest will not be asked in such cases.

But in following cases with address nodes outside building and no tags on a building and no relation connecting them - StreetComplete has no chance to work and will ask spurious quests. As such tagging is valid in Spain this quest must be disabled there.

Validity of this tagging in Spain is documented in https://wiki.openstreetmap.org/wiki/ES:Directrices_de_etiquetado_espa%C3%B1olas#N.C3.BAmeros_de_portal and confirmed here.

![screen02](https://user-images.githubusercontent.com/899988/129506667-af7fffca-bb9c-479b-8400-fd0b79bb4e49.png)
https://www.openstreetmap.org/way/395543628

**Comment by javiersanp** [comment]
That document only differ from the Addresses wiki in that establish a
preference to tag address in the entrances of the building instead of the
area. It don't have a specific mention to detached building where there
exists the options of tagging in the property entrance or in the
building/buildings entance/entrances.

The problem is to detect detached buildings which address is in nodes in
the property. If disabling the entire house numbers quest is the only
solution SC could offer, please wait before applying it until I translate
it to the Talk-es mailing list for a decision.

**Comment by javiersanp** [comment]
About the change in the title of this issue: addresses as nodes outside the building outline are valid everywhere as stated in Addresses wiki ("Tags can be added to:... on polygons representing the perimeter of the site").

**Comment by matkoniecz** [comment]
"polygons representing the perimeter of the site" is well supported, and was mentioned as

> StreetComplete works fine with address tagged on area with buildings inside

StreetComplete has problem with address nodes placed outside building (for example, nodes on perimeter of a site such as https://www.openstreetmap.org/node/6142937666 )

**Comment by javiersanp** [comment]
You are right, I was wrong.

**Comment by javiersanp** [comment]
Are landuse=residential areas supported?

**Comment by peternewman** [comment]
> But it is only possible to do this after user entered address. So after quest was asked and answered app would detect that it should not be asked in the first place and nothing would be tagged. Then it would be asked again, and again and again (even if it would be recorded locally that this is invalid target - other mappers would be asked).

> quest requirement would be impossible to satisfy. Or is there some tag to indicate "this building has address nodes placed outside it"? This would allow to at least theoretically rescue this quest.

`address=separate`? :stuck_out_tongue_winking_eye: 

But joking aside, the risk would be if the user miss-types, or either the outline has missed a number or a new one has been added, it would then tag that one unique number on the building anyway, if it was only checking for duplicates.

Also more generally, consider two streets meeting at a junction, there could easily be two house number 1's there, you're getting into some very complicated logic to not pick and use the wrong one (possibly simplified a bit if the street name has been set).

**Comment by Helium314** [comment]
> Are landuse=residential areas supported?

Any non-building areas with addresses should work: https://github.com/streetcomplete/StreetComplete/blob/master/app/src/main/java/de/westnordost/streetcomplete/quests/address/AddHousenumber.kt#L155

**Comment by peternewman** [comment]
> Any non-building areas with addresses should work

Isn't that if the area itself has an address tag.

Aren't these actually:
https://github.com/streetcomplete/StreetComplete/blob/daac414f6027e8a1c567a967250bd2a86afcab7f/app/src/main/java/de/westnordost/streetcomplete/quests/address/AddHousenumber.kt#L92-L97

Which is:
https://github.com/streetcomplete/StreetComplete/blob/6bb4e0c11ee109262d26f4473a4c14375729f3d7/app/src/main/java/de/westnordost/streetcomplete/ktx/Element.kt#L36-L61

Which obviously the wall of https://www.openstreetmap.org/way/655678691 won't currently hit. But yes, I believe `landuse=residential` would do the trick.

**Comment by javiersanp** [comment]
I mention the landuse=residential area because the problem with the detached buildings could be solved mapping the property (cadastral parcel) around it.

**Comment by peternewman** [comment]
> I mention the landuse=residential area because the problem with the detached buildings could be solved mapping the property (cadastral parcel) around it.

Yes exactly, if you tag the landuse, I think the problem will solve itself, which may give you the best of all words, usable for standalone houses still, while not incorrectly double-tagging buildings mapped like this.

**Comment by javiersanp** [comment]
In [talk-es](
https://lists.openstreetmap.org/pipermail/talk-es/2021-August/017842.html)
place=plot have been mentioned. I think it's more precise as
landuse=residential is frequently already mapped for bigger areas and it
will result in overlapping landuses with the same value.

I would insist in that this is not a local problem. In the Addresses
wiki: "Addresses
can be tagged with:...

   - nodes that are parts of building polygons ( = entrance
   <https://wiki.openstreetmap.org/wiki/Key:entrance>=*s) or on the access
   to a property (e.g. on barrier
   <https://wiki.openstreetmap.org/wiki/Key:barrier>=gate
   <https://wiki.openstreetmap.org/wiki/Tag:barrier%3Dgate> or barrier
   <https://wiki.openstreetmap.org/wiki/Key:barrier>=entrance
   <https://wiki.openstreetmap.org/wiki/Tag:barrier%3Dentrance>), used in
   places where addresses are assigned to entrances and accesses rather than
   to buildings)"

The sentence refers to the property, but it should be mapped with an area
to exists in OSM.

**Comment by matkoniecz** [comment]
> I would insist in that this is not a local problem

In many places addresses as nodes outside the building outline would be a tagging mistake (as addresses are not assigned to entrances in a given region).

Not 100% sure is Spain an unique case here (in addition to say Italy and other countries where this quest is disabled) - or is such situation happening in many other places.

**Comment by peternewman** [comment]
> In [talk-es]( https://lists.openstreetmap.org/pipermail/talk-es/2021-August/017842.html) place=plot have been mentioned.

The key `place` is also in that list, so adding that will also be sufficient to stop SC asking about it.

**Comment by westnordost** [comment]
ping

**Comment by westnordost** [comment]
There are no more replies on the Spanish mailing list and no reply here for over a month, so I'll close this ticket.
